engine/src/main/java/org/pentaho/di/trans/steps/tableoutput/TableOutputMeta.java

```diff
@@ -606,17 +606,18 @@ public void check( List<CheckResultInterface> remarks, TransMeta transMeta, Step
         remarks.add( cr );
 
         if ( !Utils.isEmpty( tableName ) ) {
+          String realSchemaName = db.environmentSubstitute( schemaName );
+          String realTableName = db.environmentSubstitute( tableName );
           String schemaTable =
```

Comment:
I think this can go or only define in case of error

Reply:
Same here. Will be touched 2-3 times at least.